### DISCUSSION TITLE
:style argument in DOM builder functions ?

### DISCUSSION THREAD

**Discussion by mmontone** [rabbibotton/clog#155] [Ideas]
I use quite a bit of inline CSS styling, but there's no `:style` argument. It is already possible to use `:class`, but I don't have every CSS as a separate rule, I want to have inline too. So, I have to resort to using `set-styles` after creating the object. That's inconvenient.

Would it be reasonable add a `style` keyword argument to DOM builder functions?

I know it is a pretty big api change, but I'd like to consider it anyway ...

**Comment by mmontone** [comment]
This is not strictly needed; I can use an html generating macro:
```lisp
(create-child cb-body (html (:div :class "w3-col s4 m4 l4" :style "height: 200px; overflow:scroll;" (who:str "No completions"))))
```
But I'm still leaving this here for consideration..

**Comment by rabbibotton** [comment]
I plan on adding, I'll try and move up to do in next days. It was not a priority since the builder takes care of all this for you, but last thing I want to see is some one feeling they need to use create-child all the time :)

**Comment by rabbibotton** [comment]
OK done, all the common element and form elements now have a :style as well.

**Reply by mmontone** [reply]
Awesome! Thanks!

**Reply by rabbibotton** [reply]
forgot to push :) there now
